### PR TITLE
terraformer planet field bonus logic

### PR DESCRIPTION
**Pull request by jackbayliss**
This should handle #392 

Also adds a variety of tests - feel free to let me know any adjustments.

### CODE REVIEW

File: app/Services/PlanetService.php [hunk 1]
```diff
@@ -353,7 +353,18 @@ public function getPlanetImageType(): string
...
-        return $this->planet->field_max;
+        $extra_fields = 0;
+        if ($this->planet->terraformer != 0) {
+            // For every level, it increases by 5
+            $extra_fields += $this->planet->terraformer * 5;
+
+            // For every 2 levels, it adds another bonus field
+            $two_level_bonus_count = (int)(floor($this->planet->terraformer / 2));
+            $extra_fields += $two_level_bonus_count;
+
```
Comment (lanedirt) [inline app/Services/PlanetService.php:364]: Nitpick: can you remove the duplicate whiteline.

File: tests/Unit/PlanetServiceTest.php [hunk 2]
```diff
@@ -121,7 +121,48 @@ public function testGetPlanetFieldMax(): void
         ]);
         $this->assertEquals(14, $this->planetService->getPlanetFieldMax());
     }
+    /**
+     * Test that the field max function with terraformer returns expected values
+     */
+    public function testGetPlanetFieldMaxWithTerraformer(): void
+    {
+
```
Comment (lanedirt) [inline tests/Unit/PlanetServiceTest.php]: Nitpick: can you remove the whiteline at the start of the method.

File: tests/Unit/PlanetServiceTest.php [hunk 3]
```diff
@@ -121,7 +121,48 @@ public function testGetPlanetFieldMax(): void
...
+        // Test none divisible by 2-- should only add 5.
+        $this->createAndSetPlanetModel([
+            'field_max' => 90,
+            'terraformer' => 1,
+        ]);
+
+        $this->assertEquals(95, $this->planetService->getPlanetFieldMax());
+
+
```
Comment (lanedirt) [inline tests/Unit/PlanetServiceTest.php]: Nitpick: can you remove the duplicate whiteline.

File: tests/Unit/PlanetServiceTest.php [hunk 4]
```diff
@@ -121,7 +121,48 @@ public function testGetPlanetFieldMax(): void
...
 
+
+        // Ensure if it's not built it doesn't alter the max fields.
+        $this->createAndSetPlanetModel([
+            'field_max' => 100,
+            'terraformer' => 0,
+        ]);
+
+        $this->assertEquals(100, $this->planetService->getPlanetFieldMax());
+    }
```
Comment (lanedirt) [inline tests/Unit/PlanetServiceTest.php:162]: Nitpick: can you add a whiteline here between the function closing brace and the phpdoc comment of the next function.

**Review by lanedirt — CHANGES_REQUESTED** [review]
Hi @jackbayliss,

Changes look good to me! I just have a few small whiteline nitpicks to make it a little bit cleaner and conform to the rest of the codebase. Could you take a look at these?

Other than these minor style tweaks, functionality looks to be working great. Thanks for your work on this! 🚀

**Comment by jackbayliss** [comment]
@lanedirt  Should hopefully now be sorted, did spot a previous test I did where I missed a whitespace after the closing brace, so sorted that too.

File: tests/Unit/PlanetServiceTest.php [hunk 5]
```diff
@@ -115,13 +115,53 @@ public function testGetPlanetFieldMax(): void
...
+        $this->createAndSetPlanetModel([
+            'field_max' => 100,
+            'terraformer' => 20,
+        ]);
+
+        // each level + 5 max fields - 100 base, plus 20*5 = 200
+        // every 2 levels + 1 max field- 20/2 = 10, so 200 + 10 = 210
+        $this->assertEquals(210, $this->planetService->getPlanetFieldMax());
+
+
```
Comment (lanedirt) [inline tests/Unit/PlanetServiceTest.php]: Nitpick: there is one duplicate whiteline remaining here.

**Review by lanedirt — CHANGES_REQUESTED** [review]
Hi @jackbayliss,

Thanks for the refactor! In the latest diff, I noticed a duplicate blank line that still remains. Could you take a look at this?

By the way, I'll try to see if I can incorporate these style checks into the PSR/PhpStan code analysis so that the build process can automatically detect them. That should make things easier going forward.

Thanks again for your work and time 😄 !

**Comment by jackbayliss** [comment]
@lanedirt  Good catch, must of missed it- I've now adjusted it.

It would be ideal if we could add this into the code analysis as I guess everyone has their own style 😃

**Review by lanedirt — APPROVED** [review]
LGTM!

**Comment by jackbayliss** [comment]
Not sure why the test failed! Looks to be a bit flaky, but I believe this is already in another issue?

**Comment by lanedirt** [comment]
> Not sure why the test failed! Looks to be a bit flaky, but I believe this is already in another issue?

Yes it looks like it's the same issue as reported in #401. Have to look into fixing that one soon. 🐛  Re-running tests now.

Edit: tests succeeded. Merging now 🚀 !

**Pull request merged**